packages/core/core/src/core-api/controller/index.ts

```diff
@@ -24,7 +25,11 @@ function createController({
 
   const proto: CoreApi.Controller.Base = {
     transformResponse(data, meta) {
-      return transformResponse(data, meta, { contentType });
+      const ctx = requestCtx.get();
+      return transformResponse(data, meta, {
+        contentType,
+        useJsonAPIFormat: ctx?.headers?.['strapi-response-format'] === 'v4',
```

Comment:
could we prefix the header with x- as it is not a standard header?

Reply:
This recommendation has been deprecated for a while now :) the strapi prefix should be perfectly fine. 

You can read more here :) https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers

Reply:
oh! deprecated since 2012 :O , alright